src/bin/tikv-ctl.rs

```diff
@@ -116,12 +130,129 @@ fn main() {
             }
         }
         dump_range(db, from, to, limit, cf_name);
+    } else if let Some(matches) = matches.subcommand_matches("mvcc") {
+        let cf_name = matches.value_of("cf").unwrap_or("default");
+        let key = matches.value_of("key").unwrap();
+        println!("You are searching Key {}: ", key);
+        match cf_name {
+            CF_DEFAULT => {
+                dump_mvcc_default(db, key);
+            }
+            CF_LOCK => {
+                dump_mvcc_lock(db, key);
+            }
+            CF_WRITE => {
+                dump_mvcc_write(db, key);
+            }
+            _ => {
+                println!("The cf: {} cannot be dumped", cf_name);
+                let _ = app.print_help();
+            }
+        }
     } else {
         let _ = app.print_help();
     }
 
 }
 
+fn gen_key_prefix(key: &str) -> Vec<u8> {
+    let encoded = encode_bytes(unescape(key).as_slice());
+    keys::data_key(encoded.as_slice())
+}
+
+pub trait MvccDeserializable {
+    fn deserialize(bytes: &[u8]) -> Self;
+}
+
+impl MvccDeserializable for Write {
+    fn deserialize(bytes: &[u8]) -> Self {
+        Write::parse(bytes).unwrap()
+    }
+}
+
+impl MvccDeserializable for Lock {
+    fn deserialize(bytes: &[u8]) -> Self {
+        Lock::parse(bytes).unwrap()
+    }
+}
+
+impl MvccDeserializable for Vec<u8> {
+    fn deserialize(bytes: &[u8]) -> Self {
+        bytes.to_vec()
+    }
+}
+
+pub struct MvccKv<T> {
+    key: Key,
+    value: T,
+}
+
+pub fn gen_mvcc_iter<T: MvccDeserializable>(db: &DB,
+                                            key_prefix: &str,
+                                            mvcc_type: CfName)
+                                            -> Option<Vec<MvccKv<T>>> {
+    let prefix = gen_key_prefix(key_prefix);
+    let mut iter = db.new_iterator_cf(mvcc_type, None, false).unwrap();
+    iter.seek(prefix.as_slice().into());
+    if !iter.valid() {
+        None
+    } else {
+        Some(iter.take_while(|s| {
+                Key::from_encoded(keys::origin_key(&s.0).to_vec())
```

Comment:
You can use map to generate `Key` first, so that it doesn't have to be copied twice.